go/vt/servenv/grpc_server_auth_static.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servenv
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+
+	log "github.com/golang/glog"
+)
+
+var (
+	credsFile = flag.String("grpc_auth_static_password_file", "", "JSON File to read the users/passwords from.")
+	// StaticAuthPlugin implements AuthPlugin interface
+	_ AuthPlugin = (*StaticAuthPlugin)(nil)
+)
+
+// StaticAuthConfigEntry is the container for server side credentials. Current implementation matches the
+// the one from the client but this will change in the future as we hooked this pluging into ACL
+// features.
+type StaticAuthConfigEntry struct {
+	Username string
+	Password string
+	// TODO (@rafael) Add authorization parameters
+}
+
+// StaticAuthPlugin  implements static username/password authentication for grpc. It contains an array of username/passwords
+// that will be authorized to connect to the grpc server.
+type StaticAuthPlugin struct {
+	entries []StaticAuthConfigEntry
+}
+
+// Authenticate implements AuthPlugin interface. This method will be used inside a middleware in grpc_server to authenticate
+// incoming requests.
+func (sa *StaticAuthPlugin) Authenticate(ctx context.Context, fullMethod string) (context.Context, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		if len(md["username"]) == 0 || len(md["password"]) == 0 {
+			return nil, grpc.Errorf(codes.Unauthenticated, "username and password must be provided")
+		}
+		username := md["username"][0]
+		password := md["password"][0]
+		for _, authEntry := range sa.entries {
+			if username == authEntry.Username && password == authEntry.Password {
+				return ctx, nil
```

Comment:
Here, if you set the `CallerID` in the context, then the username will also get validated against table ACLs downstream. Is it intentional that you didn't want this behavior?

Reply:
Yeah - I left out because would like to the ACL piece as part of the bigger revamp where we have a more holistic approach for both gRPC and tables ACL's.